lib/graphql/schema/printer.rb

```diff
@@ -7,37 +7,55 @@ class Schema
     #   MySchema = GraphQL::Schema.define(query: QueryType)
     #   puts GraphQL::Schema::Printer.print_schema(MySchema)
     #
-    module Printer
-      extend self
+    class Printer
+      attr_reader :schema, :warden
+
+      def initialize(schema, context: nil, only: nil, except: nil, warden: nil)
+        @schema = schema
+        @context = context
+        @only = only
+        @except = except
```

Comment:
Is it important to persist `@only` and `@except` in private state? If we don't _need_ them, I'd rather not save them. That way, we can avoid any mutable state bugs or untested downstream dependencies. For example, could we do something like: 

```ruby
if warden.nil?
  blacklist = build_blacklist(only, except)
  warden = # ... 
end 

@warden = warden
```